### PR TITLE
refine text in template docs.

### PR DESCRIPTION
**Pull request by lumenCodes**
As part of my outreachy contribution, I updated the [templates.md](https://github.com/jupyterhub/jupyterhub/blob/main/docs/source/reference/templates.md) file by:
- adding a section reference for easy navigation
- rephrasing some sentences to be clearer and more understandable.

### CODE REVIEW

File: docs/source/reference/templates.md [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 
 The pages of the JupyterHub application are generated from
-[Jinja](http://jinja.pocoo.org/) templates. These allow the header, for
-example, to be defined once and incorporated into all pages. By providing
-your own templates, you can have complete control over JupyterHub's
+[Jinja](http://jinja.pocoo.org/) templates. This allows the header to be defined once and incorporated into all pages. By providing your templates, you can have complete control over JupyterHub's
 appearance.
+This document will provide information on how to:
+
+- [add custom templates](#custom-templates)
```
Comment (minrk) [inline docs/source/reference/templates.md:8]: Let's not add in-body table-of-contents links. We already have these links that will be auto-generated and correct in the sidebar of the docs, whereas these links will break without warning if the spelling of a heading is changed.

File: docs/source/reference/templates.md [hunk 2]
```diff
@@ -22,16 +25,15 @@ Jinja provides a mechanism to [extend templates](http://jinja.pocoo.org/docs/2.1
...
+make extensive use of blocks, which allows you to customize parts of the interface easily.
 
 In general, a child template can extend a base template, `page.html`, by beginning with:
 
 ```html
 {% extends "page.html" %}
 ```
 
-This works, unless you are trying to extend the default template for the same
+This works if you are trying to extend the default template for the same
```
Comment (minrk) [inline docs/source/reference/templates.md:36]: This change inverts the logic making it inaccurate. It works _unless_ you do this. It _doesn't_ work _if_ you are doing it.

File: docs/source/reference/templates.md (hunk 1 above)
Comment (lumenCodes) [reply docs/source/reference/templates.md:8]: Okay, I have noted the reviews and will implement them. 

Thank you

File: docs/source/reference/templates.md (hunk 2 above)
Comment (lumenCodes) [reply docs/source/reference/templates.md:36]: okay, thanks.

**Comment by lumenCodes** [comment]
@minrk Please I have made the requested changes.

**Comment by lumenCodes** [comment]
Done @minrk

**Pull request merged**